src/firmware_logger_device.cpp

```diff
@@ -44,87 +48,99 @@ namespace librealsense
+            size_left -= log_size;
         }
     }
 
     void firmware_logger_device::get_flash_logs_from_hw_monitor()
     {
-        auto res = _hw_monitor->send(_flash_logs_command);
+        auto res = _hw_monitor->send( _flash_logs_command );
 
-        if (res.empty())
+        if( res.empty() )
         {
-            LOG_INFO("Getting Flash logs failed!");
+            LOG_INFO( "Getting Flash logs failed!" );
             return;
         }
 
-        //erasing header
+        // Erasing header
         int size_of_flash_logs_header = 27;
-        res.erase(res.begin(), res.begin() + size_of_flash_logs_header);
+        res.erase( res.begin(), res.begin() + size_of_flash_logs_header );
 
+        // Convert bytes to flash_logs_binary_data. Flash logs are supported only for legacy devices.
         auto beginOfLogIterator = res.begin();
-        // convert bytes to flash_logs_binary_data
-        for (int i = 0; i < res.size() / fw_logs::BINARY_DATA_SIZE && *beginOfLogIterator == 160; ++i)
+        for( int i = 0; i < res.size() / sizeof( fw_logs::legacy_fw_log_binary ) && *beginOfLogIterator == 160; ++i )
         {
-            auto endOfLogIterator = beginOfLogIterator + fw_logs::BINARY_DATA_SIZE;
-            std::vector<uint8_t> resultsForOneLog;
-            resultsForOneLog.insert(resultsForOneLog.begin(), beginOfLogIterator, endOfLogIterator);
+            auto endOfLogIterator = beginOfLogIterator + sizeof( fw_logs::legacy_fw_log_binary );
+            std::vector< uint8_t > resultsForOneLog;
+            resultsForOneLog.insert( resultsForOneLog.begin(), beginOfLogIterator, endOfLogIterator );
             fw_logs::fw_logs_binary_data binary_data{ resultsForOneLog };
-            _flash_logs.push(binary_data);
+            _flash_logs.push( binary_data );
             beginOfLogIterator = endOfLogIterator;
         }
 
         _flash_logs_initialized = true;
     }
 
-    bool firmware_logger_device::get_flash_log(fw_logs::fw_logs_binary_data& binary_data)
+    bool firmware_logger_device::get_flash_log( fw_logs::fw_logs_binary_data & binary_data )
     {
         bool result = false;
-        if (!_flash_logs_initialized)
+
+        if( ! _flash_logs_initialized )
         {
             get_flash_logs_from_hw_monitor();
         }
 
-        if (!_flash_logs.empty())
+        if( ! _flash_logs.empty() )
         {
-            fw_logs::fw_logs_binary_data data;
-            data = _flash_logs.front();
+            binary_data = std::move( _flash_logs.front() );
             _flash_logs.pop();
-            binary_data = data;
             result = true;
         }
+
         return result;
     }
 
-    bool firmware_logger_device::init_parser(std::string xml_content)
+    bool firmware_logger_device::init_parser( std::string xml_content )
     {
-        _parser = new fw_logs::fw_logs_parser(xml_content);
+        bool is_d400 = std::string( get_info( RS2_CAMERA_INFO_PRODUCT_LINE ) ) == "D400";
```

Comment:
Mabe we should change to switch and prepare for D600? :)
```
d400
d500
default throw
```

Reply:
OK, I will update

Reply:
I can't use `switch` because it's a string and not an integral type.
I can have an `else` that throws if you prefer it to returning a failure value.